src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2452,12 +2437,106 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
+        /// <param name="bytes">The bytes to encode.</param>
+        /// <param name="chars">The destination buffer large enough to handle the encoded chars.</param>
+        /// <param name="charLengthRequired">The pre-calculated, exact number of chars that will be written.</param>
+        private static unsafe void ToBase64CharsLargeNoLineBreaks(ReadOnlySpan<byte> bytes, Span<char> chars, int charLengthRequired)
+        {
+            // For large enough inputs, it's beneficial to use the vectorized UTF8-based Base64 encoding
+            // and then widen the resulting bytes into chars.
+            Debug.Assert(bytes.Length >= Base64VectorizationLengthThreshold);
+            Debug.Assert(chars.Length >= charLengthRequired);
+
+            // Base64-encode the bytes directly into the destination char buffer (reinterpreted as a byte buffer).
+            OperationStatus status = Base64.EncodeToUtf8(bytes, MemoryMarshal.AsBytes(chars), out _, out int bytesWritten);
+            Debug.Assert(status == OperationStatus.Done && charLengthRequired == bytesWritten);
+
+            // Now widen the ASCII bytes in-place to chars (if the vectorized ASCIIUtility.WidenAsciiToUtf16 is ever updated
+            // to support in-place updates, it should be used here instead). Since the base64 bytes are all valid ASCII, the byte
+            // data is guaranteed to be 1/2 as long as the char data, and we can widen in-place.
+            ref char dest = ref MemoryMarshal.GetReference(chars);
+            ref byte src = ref Unsafe.As<char, byte>(ref dest);
+            ref byte srcBeginning = ref src;
+
+            // We process the bytes/chars from right to left to avoid overwriting the remaining unprocessed data.
+            // The refs start out pointing just past the end of the data, and each iteration of a loop bumps
+            // the refs back the apropriate amount and performs the copy/widening.
+            dest = ref Unsafe.Add(ref dest, charLengthRequired);
+            src = ref Unsafe.Add(ref src, charLengthRequired);
+
+            // Handle 32 bytes at a time.
+            if (Vector256.IsHardwareAccelerated)
+            {
+                ref byte srcBeginningPlus31 = ref Unsafe.Add(ref srcBeginning, 31);
+                while (Unsafe.IsAddressGreaterThan(ref src, ref srcBeginningPlus31))
+                {
+                    src = ref Unsafe.Subtract(ref src, 32);
+                    dest = ref Unsafe.Subtract(ref dest, 32);
+
+                    Vector256<byte> utf8Vector = Vector256.LoadUnsafe(ref src);
+                    (Vector256<ushort> utf16Lower, Vector256<ushort> utf16Upper) = Vector256.Widen(utf8Vector);
+
+                    utf16Lower.StoreUnsafe(ref Unsafe.As<char, ushort>(ref dest));
+                    utf16Upper.StoreUnsafe(ref Unsafe.As<char, ushort>(ref dest), 16);
+                }
+            }
+
+            // Handle 16 bytes at a time.
+            if (Vector128.IsHardwareAccelerated)
```

Comment:
Could you please run the benchmarks for `Length % 32 != 0`. Sth like `1023` and benchmark it with this suggestion and without it?

I am just wondering whether it's worth keeping both `Vector256.IsHardwareAccelerated` and `Vector128.IsHardwareAccelerated`.

Currently when `Vector256` is accelerated we have to compile both Vector256 and Vector128 code paths (JIT time & code size metrics), but:
- in case `Length % 32 == 0` never execute `Vector128` path
- in case `Length % 32 != 0` the `Vector128` loop is going to be executed just once (if remaining bytes >= 16) or not at all (when < 16).

```suggestion
            }
            // Handle 16 bytes at a time.
            else if (Vector128.IsHardwareAccelerated)
```

I am just wondering if it's worth it (extra time spent in JIT + larger native code size)

Reply:
> Could you please run the benchmarks for Length % 32 != 0. Sth like 1023 and benchmark it with this suggestion and without it?

The Length in the benchmark is the input data size to Base64.EncodeToUTF8.  The output size from that is longer, and none of the lengths used here produce a multiple of 32, e.g. the Length=64 will feed a length 88 into this widening code path, which will hit the Vector256 path twice (32 + 32 = 64), the Vector128 path once (+16 = 80), and the int path twice (+4+4 = 88).

Reply:
(That said, I just realized the length produced by base64 encoding is always divisible by four, which means we don't need the one-byte-at-a-time path at all.)

Reply:
> the length produced by base64 encoding is always divisible by four

As general note / hint for base64: 
`encodedLength % 4 == 0` holds, but it can have `(0, 2]` padding chars/bytes `=`. They need to be handled specially.
For encoding: filling up to the next multiple of 4.
For decoding: stripping the `=` off before decoding the data.

I didn't follow the changes in this PR, but the one-byte-at-a-time path handles the padding in general and can't be avoided (as said I didn't follow, so if I missed something please nevermind).

Reply:
> I didn't follow the changes in this PR, but the one-byte-at-a-time path handles the padding in general and can't be avoided

Can you provide a concrete example where removing the one-byte-at-a-time path here is necessary?

Reply:
>  This code is purely about processing the resulting output of Base64.EncodeToUtf8.

Then it (potential padding) is already handled over there, so your statement about the removal is valid.

Reply:
Right.  Cool.